lwb/metalang/dynamix_runtime/dynamix_runtime/src/main/java/mb/dynamix_runtime/task/DynamixRuntimePrettyPrint.java

```diff
@@ -0,0 +1,45 @@
+package mb.dynamix_runtime.task;
+
+import mb.common.result.Result;
+import mb.dynamix_runtime.DynamixRuntimeScope;
+import mb.pie.api.ExecContext;
+import mb.pie.api.Interactivity;
+import mb.pie.api.TaskDef;
+import mb.stratego.common.StrategoException;
+import mb.stratego.common.StrategoRuntime;
+import org.spoofax.interpreter.terms.IStrategoString;
+import org.spoofax.interpreter.terms.IStrategoTerm;
+
+import javax.inject.Inject;
+import javax.inject.Provider;
+import java.util.Set;
+
+@DynamixRuntimeScope
+public class DynamixRuntimePrettyPrint implements TaskDef<IStrategoTerm, Result<String, ?>> {
+    private final Provider<StrategoRuntime> strategoRuntimeProvider;
```

Comment:
Using `DynamixRuntimeGetStrategoRuntimeProvider` here results in an exception in PIE on every build after the first, due to reasons I don't fully understand. For now this is acceptable, since the stratego ctree/jar of the runtime will be bundled with Eclipse anyway, but we should likely get a proper dependency on it.

Reply:
Can you add a TODO in the code here that we should switch to `DynamixRuntimeGetStrategoRuntimeProvider` if possible and the reason this was not done?